code/game/objects/items.dm

```diff
@@ -954,6 +957,11 @@
 
 ///Returns the sharpness of src. If you want to get the sharpness of an item use this.
 /obj/item/proc/get_sharpness()
+	var/comsig_return = SEND_SIGNAL(src, COMSIG_ITEM_SHARPNESS_OVERRIDE)
+	if (comsig_return & ITEM_SHARPNESS_OVERRIDE_EDGED)
+		return SHARP_EDGED
+	if (comsig_return & ITEM_SHARPNESS_OVERRIDE_POINTY)
+		return SHARP_POINTY
 	return sharpness
```

Comment:
Maybe should have `sharpness` as `VAR_PROTECTED` (and then add a setter so we can enforce it)